### PR TITLE
Fixed some crashes related to binding agents

### LINKED ISSUE #2311: Fix sometimes happening crashing when forming cell colonies
I don't get this crash anymore, but some people seem to do (might be framerate related):
```
ERROR: _notification: Condition "!get_viewport()" is true.
   At: scene/main/node.cpp:99
```
Note: that error message is from older Godot version, but current version still prints that "!get_viewport()" part in the error.

When adding a third cell to my colony I got this crash:
```
ERROR: _notification: Condition "!get_viewport()" is true.
   At: scene/main/node.cpp:99.
ERROR: _notification: Condition "!get_viewport()" is true.
   At: scene/main/node.cpp:99.
ERROR: _notification: Condition "!get_viewport()" is true.
   At: scene/main/node.cpp:99.
```
native stack trace:
```
	0x23c7b53 - /home/hhyyrylainen/bin/godot : _ZN9SceneTree17remove_from_groupERK10StringNameP4Node
	0x240608f - /home/hhyyrylainen/bin/godot : _ZN4Node20_propagate_exit_treeEv
	0x2405ed3 - /home/hhyyrylainen/bin/godot : _ZN4Node20_propagate_exit_treeEv
	0x2406aef - /home/hhyyrylainen/bin/godot : _ZN4Node9_set_treeEP9SceneTree
	0x240733a - /home/hhyyrylainen/bin/godot : _ZN4Node12remove_childEPS_
	0x35f1acb - /home/hhyyrylainen/bin/godot : _Z16godot_icall_1_52P10MethodBindP6ObjectS2_
```
managed stack trace:
```
	  at <unknown> <0xffffffff>
	  at Godot.NativeCalls:godot_icall_1_52 <0x000b5>
	  at Godot.Node:RemoveChild <0x000d3>
	  at NodeHelpers:Reparent <0x001cb>
	  at Microbe:ChangeNodeParent <0x000fb>
	  at Microbe:OnIGotAddedToColony <0x0037f>
	  at Microbe:OnColonyMemberAdded <0x000cf>
	  at <>c__DisplayClass19_0:<AddToColony>b__0 <0x00083>
	  at System.Collections.Generic.List`1:ForEach <0x001a5>
	  at MicrobeColony:AddToColony <0x004ab>
	  at Microbe:BeginBindWith <0x0036b>
	  at Microbe:CheckBinding <0x0036b>
	  at Microbe:OnContactBegin <0x006df>
	  at <Module>:runtime_invoke_void__this___int_object_int_int <0x00108>
	  at <unknown> <0xffffffff>
	  at Godot.NativeCalls:godot_icall_2_426 <0x000c0>
	  at Godot.Node:AddChild <0x000df>
	  at NodeHelpers:Reparent <0x0020f>
	  at Microbe:ChangeNodeParent <0x000fb>
	  at Microbe:OnIGotAddedToColony <0x0037f>
	  at Microbe:OnColonyMemberAdded <0x000cf>
	  at <>c__DisplayClass19_0:<AddToColony>b__0 <0x00083>
	  at System.Collections.Generic.List`1:ForEach <0x001a5>
	  at MicrobeColony:AddToColony <0x004ab>
	  at Microbe:BeginBindWith <0x0036b>
	  at Microbe:CheckBinding <0x0036b>
	  at Microbe:OnContactBegin <0x006df>
	  at <Module>:runtime_invoke_void__this___int_object_int_int <0x00108>
```
Perhaps this is related to (as in the node detach / re-attach ordering is causing problems): https://github.com/Revolutionary-Games/Thrive/issues/2309

I tested a fix to move the colony creation that was in the physics callback to be called through the Invoke mechanism for me that didn't fix anything but maybe that is worth another try?

### PR DESCRIPTION
**Pull request by hhyyrylainen**
**Brief Description of What This PR Does**

Trying to fix the most glaring crashing issues regarding the binding agents.
Also lays some foundations towards fixing #2029

**Related Issues**

fixes #2311

also found and fixed a bug regarding the duplicate player cheat.

**Progress Checklist**

Note: before starting this checklist the issue should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

File: src/general/IEntity.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    /// </remarks>
+    [JsonIgnore]
+    AliveMarker AliveMarker { get; }
+
+    void OnDestroyed();
+
+    // TODO: have this implementation here (and also for AliveMarker) once
+    /*{
+        AliveMarker.Alive = false;
+    }*/
```
Comment (FuJa0815) [inline src/general/IEntity.cs:25]: I don't fully understand what the TODO speciffically is here.

**Review by FuJa0815 — COMMENTED** [review]
I've got an object dispose exception once when saving during a dead microbe was on screen, but that might have been random.

I've still experianced some crashes, but this PR improved things.

Code wise this looks pretty good.

File: src/general/IEntity.cs (hunk 1 above)
Comment (hhyyrylainen) [reply src/general/IEntity.cs:25]: If I try to add that as the default implementation (C# 8 allows default implementations in interfaces now), I get an error.
But now that I re-read this, I seem to have forgotten to write the rest of the comment... I'll do that now.

**Comment by hhyyrylainen** [comment]
Probably a random disposed exception (there's a few issues open about that) as I just tried and just having dead cells on screen and they didn't cause a disposed exception when saving.

File: src/general/IEntity.cs (hunk 1 above)
Comment (hhyyrylainen) [reply src/general/IEntity.cs]: Updated

**Review by FuJa0815 — APPROVED** [review]

**Pull request merged**